DSWaveformImage/DSWaveformImage/Common/WaveformImageDrawer.swift

```diff
@@ -50,20 +50,20 @@ public class WaveformImageDrawer: ObservableObject {
     /// Renders a UIImage of the provided waveform samples.
     ///
     /// Samples need to be normalized within interval `(0...1)`.
-    public func waveformImage(from samples: [Float], with configuration: Waveform.Configuration) -> UIImage? {
+    public func waveformImage(from samples: [Float], with configuration: Waveform.Configuration) -> NSImage? {
         guard samples.count > 0, samples.count == Int(configuration.size.width * configuration.scale) else {
             print("ERROR: samples: \(samples.count) != \(configuration.size.width) * \(configuration.scale)")
             return nil
         }
 
-        let format = UIGraphicsImageRendererFormat()
-        format.scale = configuration.scale
-        let renderer = UIGraphicsImageRenderer(size: configuration.size, format: format)
-        let dampenedSamples = configuration.shouldDampen ? dampen(samples, with: configuration) : samples
-
-        return renderer.image { renderContext in
-            draw(on: renderContext.cgContext, from: dampenedSamples, with: configuration)
+      let dampenedSamples = configuration.shouldDampen ? dampen(samples, with: configuration) : samples
+      return NSImage(size: configuration.size, flipped: false) { rect in
+        guard let context = NSGraphicsContext.current?.cgContext else {
+          fatalError("Missing context")
         }
+        self.draw(on: context, from: dampenedSamples, with: configuration)
+        return true
+      }
```

Comment:
This was the only place I had to change code; everything else is just an API change which could probably be handled with a typedef.

Reply:
gotcha 🙏